lib/linzer/verifier.rb

```diff
@@ -5,12 +5,16 @@ module Verifier
     class << self
       include Common
 
-      def verify(key, message, signature)
+      def verify(key, message, signature, no_older_than: nil)
         validate message, key, signature
 
         parameters = signature.parameters
         components = signature.components
 
+        if no_older_than && (Time.now.to_i - parameters["created"]) > no_older_than.to_i
```

Comment:
If the signature being verified doesn't have a `created` parameter and is requested to check that is not older than some some value, I think we want to raise a Error, otherwise it will fail with an exception like the following:

```
TypeError: nil can't be coerced into Integer (TypeError)

          puts (Time.now.to_i - parameters["created"])
                                ^^^^^^^^^^^^^^^^^^^^^
```

Maybe in `#validate` method in `Verifier` class can be validated that if `no_older_than` option is present, then signature parameter must be present and its value should respond to `to_i`.

Reply:
Sorry, I missed that. I moved some code around and now raise a proper error in this case.